api/dbadapters/index.ts

```diff
@@ -50,7 +48,7 @@ export interface IDbAdapter {
 }
 
 export interface IDbAdapterClass<T extends IDbAdapter> {
-  create: (credentials: Credentials) => Promise<T>;
+  create: (credentials: Credentials, warehouseType?: string) => Promise<T>;
```

Comment:
```suggestion
  create: (credentials: Credentials, warehouseType: string) => Promise<T>;
```

`warehouseType` isn't optional, handling it is